### DISCUSSION TITLE
not able to create the content in course assessment

### DISCUSSION THREAD

**Discussion by srikanth-6520** [Sunbird-Ed/Community#590] [Bugs] [answered]
Hello @amith @princegupta1131

   We are trying the create the content in course assessment , but it's throwing an error , we have shared the screenshot below.
   
![image](https://github.com/Sunbird-Ed/Community/assets/133200106/37d4d6bc-c5b9-4ea3-84bb-25ba7f3effe9)

**Comment by amitpriyadarshi** [comment]
@srikanth-6520 as per the error msg obj-cat:course-assessment_content_all is not created. Please create a object category for Course Assessment and then create a definition with categoryId: obj-cat:course-assessment and targetObjectType: content
You can keep ignore objectMetadata and forms value for this definition.
cc: @Krishnaj20 @vinukumar-vs @gandham-santhosh @tsprasath

**Comment by srikanth-6520** [comment]
@amitpriyadarshi , we have already created the obj-cat for course assessment , but it's throwing error , while creating

![image](https://github.com/Sunbird-Ed/Community/assets/133200106/199b08fc-614f-4442-a428-91ecda1c27f5)

**Comment by amitpriyadarshi** [comment]
@srikanth-6520 then restart content service one.

**Reply by srikanth-6520** [reply]
@amitpriyadarshi , we have restarted the content service also , but again it is giving the same error.

**Comment by amitpriyadarshi** [comment]
@srikanth-6520 can you check in cassandra whether chategory_store.category_definition_data has entry for the respective object or not.

**Reply by srikanth-6520** [reply]
@amitpriyadarshi , we don't have the data inside the cassandra db
![image](https://github.com/Sunbird-Ed/Community/assets/133200106/ba65b14d-1576-4f21-ac82-6e00059bdc66)

**Comment by amitpriyadarshi** [comment, marked as answer]
@srikanth-6520 as per discussion with @gandham-santhosh sharing category definition update curl. Please update the respective value and run this. It will add the category definition data in cassandra.
```
curl --location --request PATCH '{{host}}/api/object/category/definition/v1/update/obj-cat:course-assessment_content_all' \
--header 'Content-Type: application/json' \
--header 'Authorization: {{api-key}}' \
--data '{
    "request": {
        "objectCategoryDefinition": {
            "objectMetadata": {
                "config": {},
                "schema": {
                    "properties": {
                        "trackable": {
                            "type": "object",
                            "properties": {
                                "enabled": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "No"
                                },
                                "autoBatch": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "No"
                                }
                            },
                            "additionalProperties": false
                        },
                        "compatibilityLevel": {
                            "type": "number",
                            "enum": [
                                5
                            ],
                            "default": 5
                        },
                        "maxAttempts": {
                            "type": "number",
                            "default": 3
                        }
                    }
                }
            },
            "forms": {}
        }
    }
}'
```
cc: @Krishnaj20 @vinukumar-vs

**Reply by srikanth-6520** [reply]
@amitpriyadarshi , now it's working fine , thank you .
